controller/state.go

```diff
@@ -136,10 +136,8 @@ func (m *appStateManager) getRepoObjs(app *v1alpha1.Application, source v1alpha1
 	if err != nil {
 		return nil, nil, nil, err
 	}
-
-	targetObjs, hooks, nil := unmarshalManifests(manifestInfo.Manifests)
-	return targetObjs, hooks, manifestInfo, nil
-
+	targetObjs, hooks, err := unmarshalManifests(manifestInfo.Manifests)
+	return targetObjs, hooks, manifestInfo, err
```

Comment:
In concert with above 🔼 
```suggestion
	return targetObjs, hooks, manifestInfo, nil
```

Reply:
why do you want to return nil here? don't you want to propagate the error?

Reply:
The error gets propagated above, see https://github.com/argoproj/argo-cd/pull/2151/files/360d9817a17c46b63de1047c69f73bef5ffc7007#r314001857